component/src/main/java/org/wso2/extension/siddhi/store/redis/RedisIterator.java

```diff
@@ -0,0 +1,286 @@
+    private List scanResultsList = new ArrayList();
+    private boolean preFetched;
+    private Object[] nextValue;
+    private boolean initialTraverse;
+    private boolean isInitialTraverse = true;
+    private BasicCompareOperation query;
+    private List<String> primaryKeys;
+    private List<String> indexes;
+    private Object[] result;
+    private StoreVariable storeVariable;
+    private StreamVariable streamVariable;
+    private Long stringCursor = RedisTableConstants.REDIS_DEFAULT_CURSOR;
+    private Iterator<Object[]> iterator;
+
+    public RedisIterator(Jedis jedis, List<Attribute> attributes, BasicCompareOperation query, String tableName,
+                         List<String> primaryKeys, List<String> indexes) {
+        this.jedis = jedis;
+        this.attributes = attributes;
+        this.tableName = tableName;
+        this.initialTraverse = true;
+        this.query = query;
+        this.primaryKeys = primaryKeys;
+        this.indexes = indexes;
+
+    }
+
+    @Override
+    public void close() throws IOException {
+        closeImpl();
+    }
+
+    private void closeImpl() {
+        if (Objects.nonNull(resultMap)) {
+            resultMap.clear();
+        }
+        result = null;
+        scanResultsList = null;
+        scanResults = null;
+    }
+
+    @Override
+    public boolean hasNext() {
+        if (!this.preFetched) {
+            // check whether object is pre fetched
+            this.nextValue = this.next();
+            this.preFetched = true;
+        }
+        return nextValue != null;
+    }
+
+    @Override
+    public Object[] next() {
+        if (this.preFetched) {
+            this.preFetched = false;
+            Object[] record = this.nextValue;
+            this.nextValue = null;
+            return record;
+        }
+        List<Object[]> finalResult;
+        if (initialTraverse) {
+            finalResult = fetchResults();
+            initialTraverse = false;
+            this.iterator = finalResult.listIterator();
+            if (finalResult.isEmpty()) {
+                this.closeImpl();
+                return null;
+            }
+        }
+        if (!iterator.hasNext()) {
+            scanResultsList.clear();
+            finalResult = fetchResults();
+            this.iterator = finalResult.listIterator();
+            if (finalResult.isEmpty()) {
+                this.closeImpl();
+                return null;
+            }
+            return iterator.next();
+        } else {
+            return iterator.next();
+        }
+
+    }
+
+    @Override
+    public void remove() {
+        //Do nothing. This is a read only iterator
+    }
+
+    private List<Object[]> fetchResults() {
+        List<Object[]> resultList = new ArrayList<>();
+
+        if (Objects.nonNull(query) &&
+                !("true".equalsIgnoreCase((query.getStreamVariable()).getName()))) {
+            return fetchResultsWithCondition();
+
+        } else if (Objects.nonNull(query) &&
```

Comment:
I think we can write this in a single line. It will not exceed the 120 limit. WDYT?

Reply:
Fixed by eef9f6d